rio_stac/stac.py

```diff
@@ -0,0 +1,135 @@
+"""Create STAC Item from a rasterio dataset."""
+
+import datetime as pydatetime
+import os
+from contextlib import ExitStack
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+import pystac
+import rasterio
+from rasterio import warp
+from rasterio.io import DatasetReader, DatasetWriter, MemoryFile
+from rasterio.vrt import WarpedVRT
+
+
+def bbox_to_geom(bbox: Sequence[float]) -> Dict:
+    """Return a geojson geometry from a bbox."""
+    # TODO: Handle dateline geometry
+    return {
+        "type": "Polygon",
+        "coordinates": [
+            [
+                [bbox[0], bbox[3]],
+                [bbox[0], bbox[1]],
+                [bbox[2], bbox[1]],
+                [bbox[2], bbox[3]],
+                [bbox[0], bbox[3]],
+            ]
+        ],
+    }
+
+
+def get_metadata(
+    source: Union[str, DatasetReader, DatasetWriter, WarpedVRT, MemoryFile]
+) -> Dict:
+    """Get Raster Metadata."""
+    metadata: Dict[str, Any] = {}
+
+    with ExitStack() as ctx:
+        if isinstance(source, (DatasetReader, DatasetWriter, WarpedVRT)):
+            src_dst = source
+        else:
+            src_dst = ctx.enter_context(rasterio.open(source))
+
+        bbox = warp.transform_bounds(
+            src_dst.crs, "epsg:4326", *src_dst.bounds, densify_pts=21
+        )
+
+        metadata["bbox"] = bbox
+        metadata["footprint"] = bbox_to_geom(bbox)
+
+        # GDAL Metadata
+        metadata["tags"] = src_dst.tags()
+        metadata["width"] = src_dst.width
+        metadata["height"] = src_dst.height
+        metadata["dtype"] = src_dst.dtypes[0]
+
+        # Proj
+        try:
+            epsg = src_dst.crs.to_epsg() or "null"
```

Comment:
Shouldn't this be `None`? You want the JSON output to have the value `null`, not `"null"`.

```py
import json
json.dumps(None)
# 'null'
json.dumps('null')
# '"null"'
```